src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3163,15 +3236,85 @@ select string.IsNullOrEmpty(function.Namespace)
         }
 #endif
 
-        private void AddNodeTypeToSearch(TypeLoadData typeLoadData)
+        private NodeModelSearchElement AddNodeTypeToSearch(TypeLoadData typeLoadData)
         {
             if (!typeLoadData.IsDSCompatible || typeLoadData.IsDeprecated || typeLoadData.IsHidden
                 || typeLoadData.IsMetaNode)
             {
-                return;
+                return null;
             }
-
+            var node = new NodeModelSearchElement(typeLoadData);
             SearchModel?.Add(new NodeModelSearchElement(typeLoadData));
+            return node;
+        }
+
+        /// <summary>
+        /// Initialize Lucene index document object for reuse
+        /// </summary>
+        /// <returns></returns>
+        private Document InitializeIndexDocument()
+        {
+            var fullCategory = new TextField("FullCategoryName", "", Field.Store.YES);
+            var name = new TextField("Name", "", Field.Store.YES);
+            var description = new TextField("Description", "", Field.Store.YES);
+            var keywords = new TextField("SearchKeywords", "", Field.Store.YES);
+            var inp = new TextField("InputParameters", "", Field.Store.YES);
+            var outp = new TextField("OutputParameters", "", Field.Store.YES);
+
+            var docName = new StringField("DocName", "", Field.Store.YES);
+            var fullDoc = new TextField("Documentation", "", Field.Store.YES);
+
+            var pkgName = new TextField("PackageName", "", Field.Store.YES);
+            var pkgVer = new TextField("PackageVersion", "", Field.Store.YES);
```

Comment:
Can you add a TODO comment to move these hard coded string to a different class?

Reply:
added comment, see commit: https://github.com/DynamoDS/Dynamo/pull/14013/commits/9af227771a8e2603c7340d32b7da53c9202362bd

Reply:
I think we can later maintain a key/value data structure, such as a Dictionary, to store field name with their value type, so that we won't have to worry about what type we used for each field.